### ISSUE TITLE
Don't pass the username when connecting to Redis < v6 even if a username is provided

### ISSUE THREAD

**Issue by jthurne** [IBM-Cloud/redli#27]
redli 0.5 added support for Redis 6 and username-based authentication. However, if redli is used to connect to an earlier version of Redis, and a username is provided, then the connection fails with:

> wrong number of arguments for 'auth' command

I know this seems silly: after all, if you don't want a username to be used, then don't pass one, but this does break certain scripts or other tools, like the IBM Cloud Databases CLI, which will pass the username regardless of the Redis version. Specifically, redli 0.5.0 and 0.5.1 do not work with when connecting to an older Redis:

```
ibmcloud cdb cxn --start MyOlderRedisDeployment INFO KEYSPACE
```

The [Python Redis library had a similar issue](https://github.com/andymccurdy/redis-py/issues/1274). To correct it, they catch the "wrong number of arguments" response from Redis and try again with only the password.  A similar fix to redli would be very much appreciated!

**Comment by ullumullu** [comment]
Thanks - this should be fixed with 0.5.2 to make usernames backwards compatible. Used the approach you linked. Very helpful!

**Comment by jthurne** [comment]
I confirmed that this is fixed with 0.5.2
